core/src/components/input/input.ionic.outline.scss

```diff
@@ -0,0 +1,162 @@
+@import "./input.vars";
+
+// Input Fill: Outline (Ionic Theme)
+// ----------------------------------------------------------------
+
+:host(.input-fill-outline) {
+  --border-radius: 8px;
+  --padding-start: 20px;
+  --padding-end: 20px;
+
+  min-height: 56px;
+}
+
+/**
+ * The bottom content should never have
+ * a border with the outline style.
+ */
+:host(.input-fill-outline) .input-bottom {
+  border-top: none;
+}
+
+/**
+ * Outline inputs do not have a bottom border.
+ * Instead, they have a border that wraps the
+ * input + label.
+ */
+:host(.input-fill-outline) .input-wrapper {
+  border-bottom: none;
+}
+
+:host(.input-fill-outline.input-label-placement-stacked) .label-text-wrapper,
+:host(.input-fill-outline.input-label-placement-floating) .label-text-wrapper {
+  @include transform-origin(start, top);
+
+  position: absolute;
+
+  /**
+   * Label text should not extend
+   * beyond the bounds of the input.
+   */
+  max-width: calc(100% - var(--padding-start) - var(--padding-end));
+}
+
+/**
+ * The label should appear on top of an outline
+ * container that overlaps it so it is always clickable.
+ */
+:host(.input-fill-outline) .label-text-wrapper {
+  position: relative;
+}
+
+/**
+ * This ensures that the input does not
+ * overlap the floating label while still
+ * remaining visually centered.
+ */
+:host(.input-fill-outline.input-label-placement-stacked) input,
+:host(.input-fill-outline.input-label-placement-floating) input {
+  @include margin(6px, 0, 6px, 0);
+}
+
+
+// Input Fill: Outline, Outline Container
+// ----------------------------------------------------------------
+
+/**
+ * Note that while the ionic theme does not make use of the notch,
+ * thus avoiding the need for separate outline container parts at all,
+ * the element's structure has been kept the same to simplify the
+ * component render function and keep selectors consistent between
+ * the ionic and md themes.
+ */
```

Comment:
I can change the render function instead if desired; this just seemed easier to follow to me.

Reply:
This works for me, it lessens the complexity of the structure if we only use styling. Plus, the comment notes what is going on.